src/main/java/ay2021s1_cs2103_w16_3/finesse/model/FinanceTracker.java

```diff
@@ -171,25 +99,35 @@ public String toString() {
 
     @Override
     public ObservableList<Expense> getExpenseList() {
-        return expenses.asUnmodifiableObservableList();
+        ObservableList<Expense> expenses = FXCollections.observableArrayList();
+        transactions.forEach(t -> {
+            if (t instanceof Expense) {
+                expenses.add((Expense) t);
+            }
+        });
+        return FXCollections.unmodifiableObservableList(expenses);
     }
 
     @Override
     public ObservableList<Income> getIncomeList() {
-        return incomes.asUnmodifiableObservableList();
+        ObservableList<Income> incomes = FXCollections.observableArrayList();
+        transactions.forEach(t -> {
+            if (t instanceof Income) {
+                incomes.add((Income) t);
+            }
+        });
+        return FXCollections.unmodifiableObservableList(incomes);
     }
```

Comment:
I feel like this could be improved somehow, due to the similar logic. I'll try to think of something

Reply:
Never mind, I think it should be fine to leave it as it is for now